swagger.yaml

```diff
@@ -2820,36 +2882,78 @@ paths:
 definitions:
   operator:
     type: object
+    required:
+      - name
+      - operator_type
     properties:
+      name:
+        type: string
+        description: Unique name of the operator.
       operator_type:
-        $ref: '#/definitions/operator-type'
-      enabled:
-        type: boolean
-        default: false
+        type: string
+        enum: ['built-in', 'custom']
+        description: Kind of operator. Different types are monitored by the service differently.
+      timeout:
+        type: integer
+        description: Positive number represents a timeout in seconds for the operator to be available.
+
+  operator-status:
+    type: object
+    required:
+      - id
+      - cluster_id
+    properties:
+      id:
```

Comment:
what id? why it is needed?

Reply:
Why not use the operator name as the ID?
+1 for no more JSON marshal messes